src/service/variable-source.js

```diff
@@ -264,4 +278,48 @@ export class VariableSource {
     }
     return new RegExp(regexStr, 'g');
   }
+
+  /**
+   * @return {?Array<string>} The whitelist of allowed AMP variables. (if provided in
+   *     a meta tag).
+   * @private
+   */
+  getVariableWhitelist_() {
+    if (this.variableWhitelist_) {
+      return this.variableWhitelist_;
+    }
+
+    const head = this.ampdoc.getRootNode().head;
+    if (!head) {
+      return null;
+    }
+
+    // A meta[name="amp-variable-substitution-whitelist"] tag, if present,
+    // contains, in its content attribute, a whitelist of variable substitution.
+    const meta =
+      head.querySelector('meta[name="amp-variable-substitution-whitelist"]');
```

Comment:
Can we rename it to `amp-allowed-url-macros` ?
- so we know it's only for URL 
- we had plan to rename `variable` to `macro` because it's not just `variable` anymore, it also has `function`

Reply:
To clarify since this is confusing: The code still says `variable` everywhere but we hope to rename everything to `macro` in the future. Since this name will be an external api (and impossible to rename later) we would like to get ahead of this by using `amp-allowed-url-macros`

Reply:
Absolutely. Done.